dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/info/NetworkInfoDeserializer.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.net.info
+
+import com.datadog.android.core.internal.domain.Deserializer
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.google.gson.JsonParseException
+import com.google.gson.JsonParser
+
+internal class NetworkInfoDeserializer : Deserializer<NetworkInfo> {
+
+    override fun deserialize(model: String): NetworkInfo? {
+        return try {
+            val jsonObject = JsonParser.parseString(model).asJsonObject
+            val connectivity = jsonObject.get(NetworkInfoSerializer.CONNECTIVITY)?.let {
+                NetworkInfo.Connectivity.valueOf(it.asString)
+            } ?: NetworkInfo.Connectivity.NETWORK_NOT_CONNECTED
+            val carrierId = jsonObject.get(NetworkInfoSerializer.CARRIER_ID).asInt
```

Comment:
Just to be more resilient, we can use `asIntOrNull` and fallback to default values.
`asInt` will crash if the value can't be converted to an int